### ISSUE TITLE
jupyter labextension install fails with: Cannot find module 'neo-async'

### ISSUE THREAD

**Issue by gnurser** [jupyterlab/jupyterlab#6367]
I am trying to install the ipympl matplotlib jupyter lab extensions.  
Centos 6.10 with python 3.6.7 from conda  
  
already installed:    
**conda install -c conda-forge jupyterlab**  
**conda install -c conda-forge ipympl**  
**conda install -c conda-forge nodejs**  
  
**$ conda list jupyterlab**
jupyterlab                0.35.6                   py36_0    conda-forge
jupyterlab_server         0.2.0                      py_0    conda-forge

**$ conda list ipympl**
ipympl                    0.2.1                 py36_1001    conda-forge

**$ conda list nodejs**
nodejs                    11.14.0              he1b5a44_1    conda-forge

Then  
**jupyter labextension install @jupyter-widgets/jupyterlab-manager**  
fails with various errors; the key one seems to be here:  

error /noc/users/agn/miniconda3/share/jupyter/lab/staging/node_modules/jupyter-matplotlib:    Command failed.  
Exit code: 1  
Command: npm run build  
Arguments:   
Directory: /noc/users/agn/miniconda3/share/jupyter/lab/staging/node_modules/jupyter-matplotlib  
Output:  
> jupyter-matplotlib@0.3.0 build /noc/users/agn/miniconda3/share/jupyter/lab/staging/node_modules/jupyter-matplotlib  
> webpack  

/noc/users/agn/miniconda3/share/jupyter/lab/staging/node_modules/webpack-cli/bin/cli.js:74  
				throw err;  
				^

Error: Cannot find module 'neo-async'  

[jupyter-labextension.txt](https://github.com/jupyterlab/jupyterlab/files/3187598/jupyter-labextension.txt)

**Comment by jasongrout** [comment]
This works fine for me in a fresh conda environment on macOS.

```
conda create -n del-jlab6367
conda activate del-jlab6367
conda install -c conda-forge jupyterlab
conda install -c conda-forge ipympl
conda install -c conda-forge nodejs
jupyter labextension install @jupyter-widgets/jupyterlab-manager
```

```
$ conda list
# packages in environment at [snip]/envs/del-jlab6367:
#
# Name                    Version                   Build  Channel
appnope                   0.1.0                 py37_1000    conda-forge
attrs                     19.1.0                     py_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
bleach                    3.1.0                      py_0    conda-forge
bzip2                     1.0.6             h1de35cc_1002    conda-forge
ca-certificates           2019.3.9             hecc5488_0    conda-forge
certifi                   2019.3.9                 py37_0    conda-forge
cycler                    0.10.0                     py_1    conda-forge
decorator                 4.4.0                      py_0    conda-forge
defusedxml                0.5.0                      py_1    conda-forge
entrypoints               0.3                   py37_1000    conda-forge
freetype                  2.10.0               h24853df_0    conda-forge
ipykernel                 5.1.1            py37h24bf2e0_0    conda-forge
ipympl                    0.2.1                 py37_1001    conda-forge
ipython                   7.5.0            py37h24bf2e0_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.4.2                      py_0    conda-forge
jedi                      0.13.3                   py37_0    conda-forge
jinja2                    2.10.1                     py_0    conda-forge
jsonschema                3.0.1                    py37_0    conda-forge
jupyter_client            5.2.4                      py_3    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
jupyterlab                0.35.6                   py37_0    conda-forge
jupyterlab_server         0.2.0                      py_0    conda-forge
kiwisolver                1.1.0            py37h770b8ee_0    conda-forge
libblas                   3.8.0               10_openblas    conda-forge
libcblas                  3.8.0               10_openblas    conda-forge
libcxx                    8.0.0                         2    conda-forge
libcxxabi                 8.0.0                         2    conda-forge
libffi                    3.2.1             h6de7cb9_1006    conda-forge
libgfortran               3.0.1                         0    conda-forge
liblapack                 3.8.0               10_openblas    conda-forge
libpng                    1.6.37               h2573ce8_0    conda-forge
libsodium                 1.0.16            h1de35cc_1001    conda-forge
markupsafe                1.1.1            py37h1de35cc_0    conda-forge
matplotlib                3.1.0                    py37_0    conda-forge
matplotlib-base           3.1.0            py37habd5b94_0    conda-forge
mistune                   0.8.4           py37h1de35cc_1000    conda-forge
nbconvert                 5.5.0                      py_0    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
nodejs                    11.14.0              h6de7cb9_1    conda-forge
notebook                  5.7.8                    py37_0    conda-forge
numpy                     1.16.3           py37hdf140aa_0    conda-forge
openblas                  0.3.6                hd44dcd8_2    conda-forge
openssl                   1.1.1b               h01d97ff_2    conda-forge
pandoc                    2.7.2                         0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.4.0                      py_0    conda-forge
pexpect                   4.7.0                    py37_0    conda-forge
pickleshare               0.7.5                 py37_1000    conda-forge
pip                       19.1                     py37_0    conda-forge
prometheus_client         0.6.0                      py_0    conda-forge
prompt_toolkit            2.0.9                      py_0    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pygments                  2.4.0                      py_0    conda-forge
pyparsing                 2.4.0                      py_0    conda-forge
pyrsistent                0.15.2           py37h01d97ff_0    conda-forge
python                    3.7.3                h0d93f26_0    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
pyzmq                     18.0.1           py37h2d07e9b_1    conda-forge
readline                  7.0               hcfe32e1_1001    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                41.0.1                   py37_0    conda-forge
six                       1.12.0                py37_1000    conda-forge
sqlite                    3.26.0            h1765d9f_1001    conda-forge
terminado                 0.8.2                    py37_0    conda-forge
testpath                  0.4.2                   py_1001    conda-forge
tk                        8.6.9             ha441bb4_1001    conda-forge
tornado                   6.0.2            py37h01d97ff_0    conda-forge
traitlets                 4.3.2                 py37_1000    conda-forge
wcwidth                   0.1.7                      py_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.33.4                   py37_0    conda-forge
widgetsnbextension        3.4.2                 py37_1000    conda-forge
xz                        5.2.4             h1de35cc_1001    conda-forge
zeromq                    4.3.1             h0a44026_1000    conda-forge
zlib                      1.2.11            h1de35cc_1004    conda-forge
```
Can you either try again, or give the exact commands to reproduce in a clean environment? Or perhaps it is centOS vs macOS. Can someone try reproducing this on a linux box?

**Comment by gnurser** [comment]
Thanks for looking at this. It seems that the problem is the version of matplotlib: v3.1 is required.   
When the jupyter labextension install failed I had (on OSX): 
``` bash
~ $ conda activate basemap
(basemap) ~ $ conda list | grep matplotlib
matplotlib                3.0.3                    py37_1    conda-forge
matplotlib-base           3.0.3            py37habd5b94_1    conda-forge
```
On OSX I could update matplotlib to v 3.1; so: 
```conda update matplotlib```
& then.   
```bash
(basemap) ~ $ conda list | grep matplotlib
matplotlib                3.1.0                    py37_0    conda-forge
matplotlib-base           3.1.0            py37habd5b94_0    conda-forge
```
... and jupyter labextension now installs OK.  
  
However on Centos, I cannot update matplotlib to v3.1:  
```bash
(basemap) ~ $ conda list | grep matplot
matplotlib                3.0.3                    py37_1    conda-forge
matplotlib-base           3.0.3            py37h5f35d83_1    conda-forge
```  
and if I try 
```conda update matplotlib```
I get: 
``` bash
(basemap) ~ $ conda update matplotlib
Collecting package metadata: done
Solving environment: done

# All requested packages already installed.
```
where still
```bash
(basemap) ~ $ conda list | grep matplot
matplotlib                3.0.3                    py37_1    conda-forge
matplotlib-base           3.0.3            py37h5f35d83_1    conda-forge
```

**Comment by jasongrout** [comment]
It looks like matplotlib 3.1 is available for linux64 as of a few days ago: https://anaconda.org/conda-forge/matplotlib/files

**Comment by gnurser** [comment]
Seems like it might be a dependency issue here on linux64; conda will install matplotlib 3.1 in a clean environment:
```
(basemap) ~ $ conda create -n matplotlib_ver
...
(basemap) ~ $ conda activate matplotlib_ver
(matplotlib_ver) ~ $ conda install matplotlib
Collecting package metadata: done
Solving environment: done

## Package Plan ##

  environment location: /noc/users/agn/miniconda3/envs/matplotlib_ver

  added / updated specs:
    - matplotlib


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    matplotlib-3.1.0           |           py37_0           6 KB  conda-forge
    matplotlib-base-3.1.0      |   py37h5f35d83_0         6.6 MB  conda-forge
 ```

**Comment by gnurser** [comment]
Hi,
Using matplotlib 3.1 sorted this problem on MacOS (High Sierra), but *not* on Centos.  
I left this for a while then discovered that first going to the jupyterlab/staging directory (essential!)   
`cd ~/miniconda3/envs/basemap/share/jupyter/lab/staging`  
and then issuing     
`npm install neo-async`  
sorted the issue.

**Comment by gnurser** [comment]
Problem is completely fixed with matplotlib 3.1.1 and jupyter_lab 1.0.4
